scout/adapter/mongo/clinvar.py

```diff
@@ -0,0 +1,127 @@
+# -*- coding: utf-8 -*-
+import logging
+from datetime import datetime
+from scout.exceptions import IntegrityError
+
+import pymongo
+
+logger = logging.getLogger(__name__)
+
+class ClinVarHandler(object):
+    """Class to handle clinvar submissions for the mongo adapter"""
+
+    def add_clinvar_submission(self, submission_obj, user, institute, case):
+        """Add a clinvar submission for a case. A submission consists of one or more variants added to the clinvar collection.
+
+            Args:
+                submission_obj(dict)
+                user_obj(dict)
+                institute_obj(dict)
+                case_obj(dict)
+
+            Returns:
+                number of documents inserted into the clinvar submissions collection (== variants submitted).
+
+        """
+        logger.info("Adding a clinvar submission '%s' for case %s", submission_obj[0]['clinvar_submission'], user)
+
+        # create one object for each submitted variant:
+        ids = []
+        for subm_variant in submission_obj:
+            subm_variant['user'] = user
+            subm_variant['institute'] = institute
+            subm_variant['case'] = case
+            subm_variant['created_at'] = datetime.now()
+            ids.append(subm_variant['_id'])
+
+        if self.clinvars_from_clinvarid(submission_id=submission_obj[0]['clinvar_submission']):
+            return "clinvar submission id "+submission_obj[0]['clinvar_submission']+" already exists in database!","danger"
+        elif self.clinvars_from_variantids(ids):
+            return "One of more variants your are trying to save is already present in a clinvar submission!","danger"
+        else:
+            logger.info("Adding clinvar submission id: {0}".format(submission_obj[0]['clinvar_submission']))
+            result = self.clinvar_collection.insert_many(submission_obj)
+            return "variants with ids {0} were saved into clinvar submission collection".format(result.inserted_ids), "success"
+
+
```

Comment:
These three functions could be collapsed to one with multiple arguments like

```python
def clinvars(self, variant_ids=None, case_id=None, submission_id=None)
    """Something here"""
    # Check that one of the parameters is used
```

Reply:
Yes, you are right, much better this way!!